src/program.ts

```diff
@@ -3398,7 +3474,29 @@ export class Function extends TypedElement {
   lookup(name: string): Element | null {
     var locals = this.localsByName;
     if (locals.has(name)) return assert(locals.get(name));
-    return this.parent.lookup(name);
+    var parentResult = this.parent.lookup(name);
+    if(parentResult && this.parent.kind == ElementKind.FUNCTION) {
+      var parentFunction = <Function>this.parent;
+      if(parentFunction.closedLocals.size > 0) { //TODO allow nested closure definitions
```

Comment:
```suggestion
      if (parentFunction.closedLocals.size > 0) { //TODO allow nested closure definitions
```

And perhaps cache `.closedLocals` in a local here since it is used again later.